python/arcticdb/storage_fixtures/utils.py

```diff
@@ -0,0 +1,105 @@
+"""
+Copyright 2023 Man Group Operations Limited
+
+Use of this software is governed by the Business Source License 1.1 included in the file LICENSE.txt.
+
+As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+"""
+
+import multiprocessing
+import subprocess
+import os
+import platform
+import requests
+import signal
+import socketserver
+import time
+import warnings
+from typing import Union, Any
+from contextlib import contextmanager
+
+_WINDOWS = platform.system() == "Windows"
+
+
+def get_ephemeral_port():  # https://stackoverflow.com/a/61685162/
+    with socketserver.TCPServer(("localhost", 0), None) as s:
+        return s.server_address[1]
+
+
+ProcessUnion = Union[multiprocessing.Process, subprocess.Popen]
+
+
+class GracefulProcessUtils:
```

Comment:
(nit) No reason for this to be a class rather than a normal Python module?

Reply:
It makes it clear the three methods should be used as a system. Otherwise, there's a higher risk that someone might call `terminate()` on a process not started by `start()` and cause a confusing mess.

Ideally, these functions become methods of a full process wrapper class, but we would end up having the `__getattr__` debate all over again.